sinonjs/sinon
# Dev dependency updates

#### Purpose (TL;DR) - mandatory

A little maintenance on dev dependencies related to our test runner setup and referee.

 #### Background (Problem in detail)  - optional

I found the latest puppeteer to be more reliable. At least upgrading in mochify made the tests pass reliably while the previous release was very flaky and had random failures in any node environment.

The latest referee introduces `assert.isUndefined` in favor of `refute.defined` so that it aligns with `assert.isNull` and friends. A deprecation warning was introduced, so I changed the assertions accordingly.

 #### How to verify - mandatory
1. Check out this branch
2. `npm ci`
3. `npm test`
4. `npm run test-coverage`
5. `npm run test-cloud`

 #### Checklist for author
